### ISSUE TITLE
Rule Request: Deinit Required

### ISSUE THREAD

**Issue by BenStaveleyTaylor** [realm/SwiftLint#2620]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

I would like to have the ability to require classes to have a `deinit` method.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

When investigating memory leaks it is useful to be able to set a breakpoint on the `deinit` method. Our house style is that all classes should have a deinit even if it is empty, to facilitate debugging. 
Frequently the class does not have a method, so you have to quit, add the method and rebuild in order to proceed. 

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
```swift
// Would trigger
class Apple {
}

class Banana
{
   func peel() {
   }
}

class Cherry: NSObject {
}

class Damson: Protocol1, Protocol2 {
}

// Would not trigger
class Apple {
   deinit {
   }
}

class Banana
{
   func peel() {
   }  
   deinit {
      print("Deinit of Banana: \(self)")
   }
}

struct Cherry {
}

enum Damson {
}

protocol Elderberry {
}
```
3. Should the rule be configurable, if so what parameters should be configurable?
No, I don't think anything beyond the standard error/warning, enable/disable is needed.

4. Should the rule be opt-in or enabled by default? Why?
opt-in. Clearly not everybody wants this all the time.

**Comment by BenStaveleyTaylor** [comment]
@marcelofabri I'd be prepared to have a go at this. Before I invest time in it, is this a rule that would be accepted if it worked as described? If the feedback is that people don't want it, that's OK.

**Comment by marcelofabri** [comment]
Implemented in #2624.
